### DISCUSSION TITLE
Roadmap and Ideas for future releases

### DISCUSSION THREAD

**Discussion by jeisenbath** [jeisenbath/ansible-collection-solarwinds-orion#42] [Ideas]
This discussion is the place to ask for new features!

Features that are on the list to develop include:
New module for managing nodes in Asset Inventory
Adding the ability to add nodes via discovery
Writing a lookup plugin

Features needing research
Updating SNMPv3 credentials for a node (currently being worked on by another contributor Andyjb8)

**Comment by Andyjb8** [comment]
Thanks for all your work on this collection! It has been a huge help in my environment.
What is your timeframe for getting this collection added to galaxy?

**Reply by jeisenbath** [reply]
There is a document outlining requirements for inclusion into the community package, I have been using those as my rubric for uploading. The major thing that I have been delaying is renaming everything to my personal namespace. Since I don't work for solarwinds, I wouldn't want to upload to galaxy in the solarwinds namespace (and solarwinds historically for orionsdk doesn't offer official dev support, I believe). The correct thing to do, likely, is put a deprecation warning on this next release to notify anyone that it will change, and create a whole new repo with the renamed collection. After 6 months then, I'd archive this repo.

Last week I was able to fix some sanity test errors, and put the env fallback into the modules and inventory plugin, made the first integration test for orion_node module, and have started an outline for a CONTRIBUTING document. Once this has more eyes on it, maintaining will be more work if I don't document all of that at the minimum, so getting integration tests and putting a guide for testing is top priority. 

And truly, thanks for using it so much. You've helped iron out a lot of the parts that I rarely use myself, so it's a big help.
